domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt

```diff
@@ -156,16 +161,18 @@ class TopicController @Inject constructor(
     topicId: String,
     storyId: String,
     explorationId: String
-  ): DataProvider<ChapterSummary> {
-    return dataProviders.createInMemoryDataProviderAsync(GET_CHAPTER_PROVIDER_ID) {
-      return@createInMemoryDataProviderAsync AsyncResult.success(
-        fetchChapter(
-          topicId,
-          storyId,
-          explorationId
+  ): LiveData<AsyncResult<ChapterSummary>> {
+    return MutableLiveData(
+      try {
+        AsyncResult.success(fetchChapter(topicId, storyId, explorationId))
+      } catch (e: Exception) {
+        AsyncResult.failed(
+          ChapterNotFoundException(
+            "Chapter for exploration $explorationId not found in story $storyId and topic $topicId"
+          )
```

Comment:
If you need a failure case then I suggest returning AsyncResult.failure() in the original DataProvider.

Reply:
updated retrieveChapter